### PR TITLE
feat(island-ui): add new popover variant for filter

### PR DESCRIPTION
**Pull request by stjanilofts**
Adds a new `popover` variant to `Filter` so now there are 3:
- `default`: The normal filter view.
- `popover`: A dropdown that opens over the content below.
- `dialog`: The same as the `isDialog` option as before.

## Screenshots / Gifs

recording

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by kksteini — APPROVED** [review]
LGTM

**Review by siggifv — APPROVED** [review]
LGTM

File: libs/island-ui/core/src/lib/Filter/Filter.stories.tsx [hunk 1]
```diff
@@ -67,28 +68,30 @@ export const Default = () => {
 
   const renderDefaultFilter = (asDialog?: boolean) => (
     <Filter
+      labelClearAll="Hreinsa allar síu"
```
Comment (oddsson) [inline libs/island-ui/core/src/lib/Filter/Filter.stories.tsx:71]:
```suggestion
      labelClearAll="Hreinsa allar síur"
```

File: libs/island-ui/core/src/lib/Filter/Filter.tsx [hunk 2]
```diff
@@ -38,29 +48,95 @@ export interface FilterProps {
...
+    unstable_flip: true,
+    gutter: 8,
+  })
+
+  const hasFilterInput = !!filterInput
 
   return (
-    <FilterContext.Provider value={{ isDialog }}>
-      {isDialog ? (
+    <FilterContext.Provider value={{ variant }}>
```
Comment (oddsson) [inline libs/island-ui/core/src/lib/Filter/Filter.tsx:82]: Could you use the `useContext` hook here, like you do in libs/island-ui/core/src/lib/Filter/FilterMultiChoice/FilterMultiChoice.tsx

**Review by oddsson — CHANGES_REQUESTED** [review]

**Review by oddsson — APPROVED** [review]

**Pull request merged**